lib/reline/line_editor.rb

```diff
@@ -1673,24 +1673,22 @@ def finish
           finish
         end
       else
-        if @line_index == (@buffer_of_lines.size - 1)
-          if confirm_multiline_termination
-            finish
-          else
-            key_newline(key)
-          end
-        else
-          # should check confirm_multiline_termination to finish?
-          @line_index = @buffer_of_lines.size - 1
-          @byte_pointer = current_line.bytesize
+        if @line_index == @buffer_of_lines.size - 1 && confirm_multiline_termination
           finish
+        else
+          key_newline(key)
         end
       end
     else
       finish
     end
   end
 
+  private def ed_force_newline(_key)
```

Comment:
It feels like force_newline force enter newline character instead of finish for me.
How about a name like force_finish or force_submit (the word submit is not use in reline, though)

Reply:
Sure. I have renamed `ed_force_submit`. Since this is a name that users can use, I thought `submit` would be easier to understand.